### PR TITLE
Automate API documentation and necessary changes for 3.9 version

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR adds:
* A new automated script to generate vagrant based environment to generate documentation.
* Necessary changes for the documentation in 3.8 version.

Best regards,
Marta

### CODE REVIEW

File: doc/build_environment/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+n_workers = 1
+n_agents = 1
+
+Vagrant.configure("2") do |config|
+
+    # Manager master
+    config.vm.define "master" do |node|
+        node.vm.box = "ubuntu/bionic64"
+        node.vm.synced_folder "C:\\Users\\Marta\\Documents\\GitHub", "/home/vagrant/GitHub"
```
Comment (crd1985) [inline doc/build_environment/Vagrantfile]: Please, remove hardcoded paths :)

**Comment by crd1985** [comment]
It would be great to install sphinx inside the vm and get the documentation compiled and ready to use.

File: doc/build_environment/Vagrantfile (hunk 1 above)
Comment (mgmacias95) [reply doc/build_environment/Vagrantfile]: Done!

**Comment by mgmacias95** [comment]
Hello team,

I have added necessary changes to build https://github.com/wazuh/wazuh-documentation/pull/938 in this PR.

Best regards,
Marta

**Pull request merged**